File: src/main/java/nz/co/manager/api/SeasonalVariation.java
package nz.co.manager.api;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;

import nz.co.manager.jdbi.Idable;

@Entity
@Table(name = "seasonalvariation")
public class SeasonalVariation extends Idable {
	@Column
	private String variation;
	@Column
	private int min;
	@Column
	private int max;

	public String getVariation() {
		return variation;
	}

	public void setVariation(final String name) {
		this.variation = name;
	}

	public int getMin() {
		return min;
	}

	public void setMin(final int min) {
		this.min = min;
	}

	public int getMax() {
		return max;
	}

	public void setMax(final int max) {
		this.max = max;
	}
}

File: src/main/resources/js/components/seasonalVariation.js
define(['knockout'], function(ko) {
	return function SeasonalVariation(data) {
		var self = this;
		if (data != null) {
			self.id = data.id;
			self.variation = ko.observable(data.variation);
			self.min = ko.observable(data.min);
			self.max = ko.observable(data.max);
		} else {
			self.variation = ko.observable();
			self.min = ko.observable();
			self.max = ko.observable();
		}
	}
});